[Bugs] unit tests fail when using go1.24 or later

**Describe the bug**
> A clear and concise description of what the bug is.

The behavior of `debug.ReadBuildInfo` has changed since 1.24, `Test_GetVersion` [tests](https://github.com/go-to-k/cls3/blob/f3ac6c57a148a520e68f281e907d83d9d929ab36/internal/version/version_test.go#L86-L103) in version_test.go get `got = "(devel)", want ""`

```go
func GetVersion() string {
	if Version != "" && Revision != "" {
		return Version + "-" + Revision
	}
	if Version != "" {
		return Version
	}

	i, ok := debug.ReadBuildInfo()
	if !ok {
		return "unknown"
	}
	return i.Main.Version
}
```

```go
func Test_GetVersion(t *testing.T) {
	type args struct {
		Version  string
		Revision string
	}

	cases := []struct {
		name    string
		args    args
		want    string
		wantErr bool
	}{
		{
			name: "Both Version and Revision are empty",
			args: args{
				Version:  "",
				Revision: "",
			},
			want:    "",
			wantErr: false,
		},
		{
			name: "Version is empty and Revision is not empty",
			args: args{
				Version:  "",
				Revision: "abcde",
			},
			want:    "",
			wantErr: false,
		},
		...
	}

	for _, tt := range cases {
		t.Run(tt.name, func(t *testing.T) {
			Version = tt.args.Version
			Revision = tt.args.Revision
			got := GetVersion()

			if got != tt.want {
				t.Errorf("got = %#v, want %#v", got, tt.want)
			}
		})
	}
}
```

It seems that it is a bug in golang. For now, I'll just comment it out.

(related)

- https://github.com/golang/go/issues/72877
- https://github.com/golang/go/issues/74763

**To Reproduce**

1. Upgrade go version to 1.24 or later in your local environment
2. `make test`

**Current workaround**

```
=== RUN   Test_GetVersion
=== RUN   Test_GetVersion/Both_Version_and_Revision_are_empty
    version_test.go:131: got = "(devel)", want ""
=== RUN   Test_GetVersion/Version_is_empty_and_Revision_is_not_empty
    version_test.go:131: got = "(devel)", want ""
=== RUN   Test_GetVersion/Revision_is_not_empty_and_Revision_is_empty
=== RUN   Test_GetVersion/Both_Version_and_Revision_are_not_empty
--- FAIL: Test_GetVersion (0.00s)
    --- FAIL: Test_GetVersion/Both_Version_and_Revision_are_empty (0.00s)
    --- FAIL: Test_GetVersion/Version_is_empty_and_Revision_is_not_empty (0.00s)
    --- PASS: Test_GetVersion/Revision_is_not_empty_and_Revision_is_empty (0.00s)
    --- PASS: Test_GetVersion/Both_Version_and_Revision_are_not_empty (0.00s)
FAIL
coverage: 0.6% of statements in ./...
FAIL    github.com/go-to-k/cls3/internal/version        0.431s
```

**Expected behavior**

The tests will not fail.

--- a/internal/version/version_test.go
+++ b/internal/version/version_test.go
@@ -83,24 +83,29 @@ func Test_GetVersion(t *testing.T) {
 		want    string
 		wantErr bool
 	}{
-		{
-			name: "Both Version and Revision are empty",
-			args: args{
-				Version:  "",
-				Revision: "",
-			},
-			want:    "",
-			wantErr: false,
-		},
-		{
-			name: "Version is empty and Revision is not empty",
-			args: args{
-				Version:  "",
-				Revision: "abcde",
-			},
-			want:    "",
-			wantErr: false,
-		},
+		// NOTE: The behavior of `debug.ReadBuildInfo` has changed since go1.24,
+		// these tests get `got = "(devel)", want ""`
+		// so the following cases are commented out
+		// TODO: Uncomment when fixed
+		// See: https://github.com/go-to-k/cls3/issues/364
+		// {
+		// 	name: "Both Version and Revision are empty",
+		// 	args: args{
+		// 		Version:  "",
+		// 		Revision: "",
+		// 	},
+		// 	want:    "",
+		// 	wantErr: false,
+		// },
+		// {
+		// 	name: "Version is empty and Revision is not empty",
+		// 	args: args{
+		// 		Version:  "",
+		// 		Revision: "abcde",
+		// 	},
+		// 	want:    "",
+		// 	wantErr: false,
+		// },
 		{
 			name: "Revision is not empty and Revision is empty",
 			args: args{

